docs/reference_guides/model_libraries/generic/costing/process_costing.rst

```diff
@@ -0,0 +1,865 @@
+    from idaes.models.unit_models import StoichiometricReactor, Flash, CSTR
+
+    # Map unit models to unit classes
+    # Will pass to unit_mapping which calls costing methods based on unit class
+    unit_class_mapping = {
+        m.fs.R101: StoichiometricReactor,
+        m.fs.F101: Flash,
+        m.fs.F102: Flash,
+        m.fs.R102: CSTR}
+
+    # Costing for vessels - m.fs.R101, m.fs.F101, m.fs.F102, m.fs.R102
+
+    # Loop over units
+    for unit in [m.fs.R101, m.fs.F101, m.fs.F102, m.fs.R102]:
+        # Get correct unit class for unit model
+        unit_class = unit_class_mapping[unit]
+        
+        # Add dimension variables and constraint if they don't exist
+        if not hasattr(unit, "diameter"):
+            unit.diameter = Var(initialize=1, units=pyunits.m)
+        if not hasattr(unit, "length"):
+            unit.length = Var(initialize=1, units=pyunits.m)
+        if hasattr(unit, "volume"):  # if volume exists, set diameter from volume
+            unit.volume_eq = Constraint(expr=unit.volume[0] == unit.length * unit.diameter**2 * 0.25 * Constants.pi)
+        else:  # fix diameter directly
+            unit.diameter.fix(0.2214 * pyunits.m)
+        # Either way, fix L/D to calculate L from D
+        # chose constant L/D of 3 for this example - this could also be a dictionary indexed by vessel/unit model
+        unit.L_over_D = Constraint(expr=unit.length == 3 * unit.diameter)
+            
+        # Define vessel costing
+        unit.costing = UnitModelCostingBlock(
+            default={
+                "flowsheet_costing_block": unit.parent_block().costing,  # e.g. m.fs.R101.costing
+                "costing_method": SSLWCostingData.unit_mapping[unit_class],  # e.g. cost_vertical_vessel()
+                "costing_method_arguments": {
+                    "material_type": VesselMaterial.CarbonSteel,
+                    "shell_thickness": 1.25 * pyunits.inch,
+                    
+                }
+            }
+        )
+
+Heat Exchanger Cost
+^^^^^^^^^^^^^^^^^^^
+
+The purchse cost is computed based on the base unit cost and three correction factors (Eq. 22.43 in Seider et al.). The base cost is computed depending on the heat exchanger type selected by the user:
+
+.. math:: self.costing.purchase\_cost = pressure\_factor*material\_factor*L\_factor*self.costing.base\_cost*(CE_{index}/500)
+
+.. math:: self.costing.base\_cost\_per\_unit = \exp{(\alpha_{1} - \alpha_{2}*\log{area*hx\_os} + \alpha_{3}*(\log{area*hx\_os})^{2})}
+
+.. math:: self.costing.base\_cost = self.costing.base\_cost\_per\_unit * self.costing.number\_of\_units
+
+.. math:: area  = self.area / self.costing.number\_of\_units
+
+where:
+
+* pressure_factor - is the pressure design correction factor
+* material_factor - is the construction material correction factor
+* length_factor - is the tube length correction factor
+* CE_index - is a global parameter for Chemical Enginering cost index for years 2010-2019
```

Comment:
The CE index is not part of the costing calculation, and should not appear here. The CE index itself is used for unit conversions, and is handled by Pyomo units.

The reason we see `USD_CE500` showing up is to add units to the expression - don't show this in the equations, but as the units associated with the purchase cost.